src/stock-adjustment-mobile/program-select.jsx

```diff
@@ -49,21 +49,45 @@ const ProgramSelect = ({ offlineService, stockReasonsFactory, existingStockOrder
                 dispatch(setProductOptions(productOptions));
                 return productOptions;
             }).then(productOptions => {
-                stockReasonsFactory.getAdjustmentReasons(program.id, facility.type.id).then(reasons => {
-                    const mappedReasons = _.map(reasons, reason => ({ name: reason.name, value: reason }));
-                    dispatch(setReasons(mappedReasons));
-                    return mappedReasons
-                }).then(mappedReasons => {
-                    if (programSelected.programId !== program.id) {
-                        dispatch(resetAdjustment(adjustment));
-                    }
-                    dispatch(setProgram(programObject));
-                    history.push(`/make${adjustmentType}AddProducts`);
-                });   
+                let reasonType = adjustmentType === "Adjustment" ? null : (adjustmentType === "Issue" ? "DEBIT" : "CREDIT" ); 
+                if (!reasonType) {
+                    stockReasonsFactory.getAdjustmentReasons(program.id, facility.type.id).then(reasons => {
+                        const mappedReasons = _.map(reasons, reason => ({ name: reason.name, value: reason }));
+                        dispatch(setReasons(mappedReasons));
+                        return mappedReasons
+                    }).then(mappedReasons => {
+                        goToProductPage(programId, programObject, program);
+                    });
+                }
+                else {
+                    stockReasonsFactory.getReasons(program.id, facility.type.id, reasonType).then(reasons => {
+                        const mappedReasons = _.map(reasons, reason => ({ name: reason.name, value: reason }));
+                        dispatch(setReasons(mappedReasons));
+                        return mappedReasons
+                    }).then(mappedReasons => {
+                        goToProductPage(programId, programObject, program);
+                    });
+                }   
             });
         }
     };
 
+    const goToProductPage = (programId, programObject, program) => {
+        sourceDestinationService.getDestinationAssignments(
+            programId, facility.id
+        ).then(sourceDestinations => {
```

Comment:
Formatting looks bad. I think
```
sourceDestinationService.getDestinationAssignments(programId, facility.id).then(sourceDestinations => {
```
is just fine

Reply:
fixed